.github/workflows/clang-tidy.yml

```
@@ -0,0 +1,45 @@
+name: "Clang-Tidy"
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+    paths:
+      - ".github/automation/x64/**"
+      - ".github/workflows/clang-tidy.yml"
+      - "src/cpu/x64/**"
+      - "src/gpu/intel/**"
```

We need a bit wider set of sources here :
```suggestion
      - cmake/**
      - examples/**
      - include/**
      - src/common/**
      - src/cpu/*
      - src/cpu/gemm/**
      - src/cpu/matmul/**
      - src/cpu/reorder/**
      - src/cpu/rnn/**
      - src/cpu/x64/**
      - tests/**
      - CMakeLists.txt
```